jerry-debugger/jerry-client-ws.py

```diff
@@ -0,0 +1,676 @@
+        else:
+            set_breakpoint(self.debugger, args)
+
+    def do_break(self, args):
+        """ Insert breakpoints in the given lines """
+        self.insert_breakpoint(args)
+
+    def do_b(self, args):
+        """ Insert breakpoints in the given lines """
+        self.insert_breakpoint(args)
+
+    def exec_command(self, args, command_id):
+        self.stop = True
+        if args != '':
+            print('Error: no argument expected')
+        else:
+            self.debugger.send_command(command_id)
+
+    def do_continue(self, args):
+        """ Continue execution """
+        self.exec_command(args, JERRY_DEBUGGER_CONTINUE)
+
+    def do_c(self, args):
+        """ Continue execution """
+        self.exec_command(args, JERRY_DEBUGGER_CONTINUE)
+
+    def do_step(self, args):
+        """ Next breakpoint, step into functions """
+        self.exec_command(args, JERRY_DEBUGGER_STEP)
+
+    def do_s(self, args):
+        """ Next breakpoint, step into functions """
+        self.exec_command(args, JERRY_DEBUGGER_STEP)
+
+    def do_next(self, args):
+        """ Next breakpoint in the same context """
+        self.exec_command(args, JERRY_DEBUGGER_NEXT)
+
+    def do_n(self, args):
+        """ Next breakpoint in the same context """
+        self.exec_command(args, JERRY_DEBUGGER_NEXT)
+
+    def do_list(self, args):
+        """ Listed the available breakpoints """
+        if args != '':
+            print('Error: no argument expected')
+            return
+
+        for breakpoint in self.debugger.active_breakpoint_list.values():
+            source = breakpoint.function.source
+            print('%d: %s' % (breakpoint.active_index, breakpoint.to_string()))
+
+    def do_delete(self, args):
+        """ Delete the given breakpoint """
+        if not args:
+            print('Error: breakpoint index expected')
+            return
+
+        try:
+            breakpoint_index = int(args)
+        except:
+            print('Error: integer number expected')
+            return
+
+        if breakpoint_index in self.debugger.active_breakpoint_list:
+            breakpoint = self.debugger.active_breakpoint_list[breakpoint_index]
+            del self.debugger.active_breakpoint_list[breakpoint_index]
+            breakpoint.active_index = -1
+            self.debugger.send_breakpoint(breakpoint)
+        else:
+            print('Error: breakpoint %d not found' % (breakpoint_index))
+
+    def exec_backtrace(self, args):
+        max_depth = 0
+
+        if args:
+            try:
+                max_depth = int(args)
+                if max_depth <= 0:
+                    print('Error: positive integer number expected')
+                    return
+            except:
+                print('Error: positive integer number expected')
+                return
+
+        message = pack(self.debugger.byte_order + 'BBIB' + self.debugger.idx_format,
+                       0x82,
```

Comment:
and what do they mean? Could we have extracted as a global variable like the JERRY_ things?